src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -3,12 +3,51 @@
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
 import core.basesyntax.model.User;
+import java.util.Objects;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_LENGTH = 6;
+    private static final int MIN_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
-    public User register(User user) {
-        return null;
+    public User register(User user) throws UserInvalidDataException {
+        nullChecking(user);
+        userMeetsCriteriaCheck(user);
+        return storageDao.add(user);
+    }
+
+    private void nullChecking(User user) throws UserInvalidDataException {
+        if (Objects.isNull(user)) {
+            throw new UserInvalidDataException("User data can't be null");
+        }
+        if (user.getLogin() == null) {
+            throw new UserInvalidDataException("User login can't be null");
+        }
+        if (user.getAge() == null) {
+            throw new UserInvalidDataException("User age can't be null");
+        }
+        if (user.getPassword() == null) {
+            throw new UserInvalidDataException("User password can't be null");
+        }
+    }
+
+    private void userMeetsCriteriaCheck(User user) throws UserInvalidDataException {
```

Comment:
this method does too much, consider creating separate methods due to my comment above